src/features/hotspots/settings/WifiSettings.tsx

```diff
@@ -34,45 +34,48 @@ const WifiSettings = ({ onNetworkSelected }: Props) => {
   const { showOKCancelAlert } = useAlert()
   const [networks, setNetworks] = useState<string[]>([])
   const [configuredNetworks, setConfiguredNetworks] = useState<string[]>([])
+  const [loading, setLoading] = useState(true)
 
   const scanWifi = useCallback(async () => {
     const wifiNetworks = await scanForWifiNetworks()
     const configured = await scanForWifiNetworks(true)
     animateTransition()
     setNetworks(wifiNetworks || [])
     setConfiguredNetworks(configured || [])
+    setLoading(false)
   }, [scanForWifiNetworks])
 
+  const maybeForgetConfiguredNetwork = useCallback(
+    async (networkToForget?: string) => {
+      if (!networkToForget) return
+      const decision = await showOKCancelAlert({
+        titleKey: 'hotspot_setup.disconnect_dialog.title',
+        messageKey: 'hotspot_setup.disconnect_dialog.body',
+        messageOptions: { wifiName: networkToForget },
+        okKey: 'generic.forget',
+      })
+
+      if (!decision) {
+        return
+      }
+      await removeConfiguredWifi(networkToForget)
+    },
+    [removeConfiguredWifi, showOKCancelAlert],
+  )
+
   const handleNetworkSelected = useCallback(
     async (nextWifi: string) => {
-      if (wifi || configuredNetworks.length) {
-        const wifiName = wifi || configuredNetworks[0]
-        const decision = await showOKCancelAlert({
-          titleKey: 'hotspot_setup.disconnect_dialog.title',
-          messageKey: 'hotspot_setup.disconnect_dialog.body',
-          messageOptions: { wifiName },
-          okKey: 'generic.forget',
-        })
-        if (!decision || !wifi) {
```

Comment:
So was the main problem that the `removeConfiguredWifi` was getting skipped?

Reply:
it was just returning early because `wifi` was empty